src/AppInstallerCLICore/ExecutionContext.cpp

```diff
@@ -128,9 +128,41 @@ namespace AppInstaller::CLI::Execution
         {
             clone->EnableCtrlHandler();
         }
+        CopyArgsToSubContext(clone.get());
         return clone;
     }
 
+    void Context::CopyArgsToSubContext(Context* subContext)
+    {
+        // Copy over install behavior flags from the parent context.
+        // Arguments not copied: Override, Log
+        for (auto flag : {

```

Comment:
Do we need to copy over some other behavior args like DisableInteractivity, Force, AcceptPackageAgreements, etc? I feel this needs a more thorough revist.

I also feel this selective copying is hard to maintain but I cannot find a better way at the moment.

Reply:
DisableInteractivity and AcceptPackageAgreements don't need to be copied over because those are handled at the top level. Unless we have nested contexts...

I've been thinking we can add some tags/flags about what an argument is for. Like, IsInstallBehavior, IsGlobalBehavior, IsQueryArgument. Then we could do this copying based on that, and we could also use it for argument validation. For example, Manifest cannot be used with query/source arguments, and installer specific args (like override) cannot be used when dealing with multiple packages (update --all)

Reply:
I just went over all the args, I suggest to copy over Interactive, Silent, Locale, InstallScope, InstallArchitecture, HashOverride, IgnoreLocalArchiveMalwareScan, NoUpgrade,  Force

Regarding InstallLocation, I saw in PromptForMultiplePackages we set them all to same value already, but I feel it will be more likely to cause conflicts for packages if installerss write to the InstallLocation root. Perhaps we should add a todo somewhere to revisit this. And this copy is probably not needed.

Reply:
@florelis / @yao-msft; Regarding this comment -
> DisableInteractivity and AcceptPackageAgreements don't need to be copied over because those are handled at the top level. Unless we have nested contexts...

I just stumbled across this issue:
- #2680

Perhaps the "top-level" flags *do* need to be copied over?